lib/ui/proposals/details/components/proposal_details_view.dart

```diff
@@ -62,312 +56,365 @@ class _ProposalDetailsViewState extends State<ProposalDetailsView> {
                       ),
                     ),
-                  ),
-                  const HyphaDivider(),
-                  /// Expiration Timer
-                  Padding(
-                    padding: const EdgeInsets.symmetric(vertical: 20),
-                    child: ProposalExpirationTimer(
-                      _proposalDetailsModel.formatExpiration(),
-                    ),
-                  ),
-                  const HyphaDivider(),
-                  /// Vote Section
-                  Padding(
-                    padding: const EdgeInsets.symmetric(vertical: 20),
-                    child: Text(
-                      'Cast your Vote',
-                      style: context.hyphaTextTheme.smallTitles,
-                    ),
-                  ),
-                  ...List.generate(
-                    3,
-                        (index) => Padding(
-                      padding: const EdgeInsets.symmetric(vertical: 10),
-                      child: HyphaAppButton(
-                        title: index == 0
-                            ? 'Yes'
-                            : index == 1
-                            ? 'Abstain'
-                            : 'No',
-                        onPressed: () async {},
-                        buttonType: ButtonType.danger,
-                        buttonColor: index == 0
-                            ? HyphaColors.success
-                            : index == 1
-                            ? HyphaColors.lightBlack
-                            : HyphaColors.error,
+                    ...List.generate(
+                      3,
+                          (index) => Padding(
+                        padding: const EdgeInsets.symmetric(vertical: 10),
+                        child: HyphaAppButton(
+                          title: index == 0
+                              ? 'Yes'
+                              : index == 1
+                              ? 'Abstain'
+                              : 'No',
+                          onPressed: () async {},
+                          buttonType: ButtonType.danger,
+                          buttonColor: index == 0
+                              ? HyphaColors.success
+                              : index == 1
+                              ? HyphaColors.lightBlack
+                              : HyphaColors.error,
 
+                        ),
                       ),
                     ),
-                  ),
-                  const SizedBox(height: 20),
-                ],
-              ),
-            );
-          });
-        },
+                    const SizedBox(height: 20),
+                  ],
+                ),
+              );
+            });
+          },
+        ),
       ),
+    );
+  }
+}
+
+Widget _buildTokenRow(
+    BuildContext context,
+    ProposalDetailsModel proposalDetailsModel,
+    int tokenIndex,
+    String tokenType,
+    bool? isShown) {
```

Comment:
isShown is kind of the wrong name for this parameter

All of this is really funky, I would refactor it with these objectives

1 - Make it clear what is happening and why

2 - Make the calculations explicit and clear, numbers should be calculated with numbers and only converted to strings at display time, by number formatters. 

3 - If there's different types involved, use enums, not plain strings based on UX text.